homeassistant/helpers/event.py

```diff
@@ -110,20 +113,20 @@ class TrackTemplateResult:
 
 
 def threaded_listener_factory(
-    async_factory: Callable[..., Any]
-) -> Callable[..., CALLBACK_TYPE]:
+    async_factory: Callable[Concatenate[HomeAssistant, _P], Any]  # type: ignore[misc]
+) -> Callable[Concatenate[HomeAssistant, _P], CALLBACK_TYPE]:  # type: ignore[misc]
     """Convert an async event helper to a threaded one."""
 
     @ft.wraps(async_factory)
-    def factory(*args: Any, **kwargs: Any) -> CALLBACK_TYPE:
+    def factory(
+        hass: HomeAssistant, *args: _P.args, **kwargs: _P.kwargs
+    ) -> CALLBACK_TYPE:
         """Call async event helper safely."""
-        hass = args[0]
```

Comment:
That's a bit strange. I had to look at that one a few times.

Reply:
Me too 🤣 !